### PR TITLE
[Commencement] Event theming.

### LINKED ISSUE #7592: Theme commencement event/ceremony nodes
# Story

As a visitor to commencement ceremony pages, I can see a fully themed page with the most important information first.

# Additional Information

Date,Venue,Body is important info that should be easy to get to. Details can be long in some cases. Should output a message with the time that says times listed are in Central Time.

Manually added ceremony information is important and should be able to get to easily.

Livestream is important and should be easy to get to.

No photo for now.

Program is tied to session so if a session is set for a ceremony, the appropriate program for that session should be displayed. The session does not need to output on the ceremony.

Don't need to display college or other celebration info on ceremony page.

Possibly work in presidential seal if possible (Josh mentioned he liked that on the current site at one point). https://brand.uiowa.edu/logos/secondary-logos#seal

```[tasklist]
### Tasks
- [x] Output date - start time & end time - output note about times are in Central Time
- [x] Link to venue
- [x] Output room if exists as part of address
- [x] Address from Venue
- [x] Output body - can get long
- [x] Do not display label for Ceremony information collections field and display this field after Ceremony information field
- [x] Display Ceremony information field label as h3 - make sure proper heading structure
- [x] Livestream info
- [x] Order of events file upload
- [x] Display program on ceremony
- [x] event link to Events Calendar
- [x] Contact name, email, phone
- [x] Website
- [x] Work in presidential seal somewhere
```

### PR DESCRIPTION
**Pull request by GaryRidgway**
tracks https://github.com/uiowa/uiowa/issues/7592

## To test

- Pull, sync commencement
- Poke around on some events and check for the following things:  

### Two column main content
- Body field properly renders
- Contact name, email, and phone properly render

### Two column sidebar card
-  Presidential seal outputs in the media slot in the sidebar.
- Date is output in sidebar - start time & end time - there is a note about times are in Central Time.
- Venue is output in sidebar with room when there is one.
- 'Program' is displayed in sidebar without pdf icon and links to the program for the selected session.  
    - (Note: you will have to go to the session taxonomy term referenced by the event and add a session PDF there for it to show up on the rendered node)
-  'Order of events' is displayed in sidebar without pdf icon.
- The Website field renders in the sidebar
    - (Note: I was unable to find any ceremony with a website, so I think this works based upon the manage display page for the layout of the event content type)
-  There is a 'View on the events site' button that links to the event's page on events.uiowa.edu.

### Livestream

- The livestream section is on a gray background and has a heading that can be linked to with `#livestream`. Additionally, it should be properly structured in the order of headings (eg. h1, h2, h3, etc).
- Livestream field properly renders

### Livestream

- The Ceremony information section has a heading that can be linked to with `#ceremony-information`. Additionally, it should be properly structured in the order of headings (eg. h1, h2, h3, etc).
- Livestream field properly renders
- Ceremony information properly renders
- Ceremony information details properly renders

### CODE REVIEW

File: docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module [hunk 1]
```diff
@@ -86,6 +120,44 @@ function commencement_core_form_alter(&$form, FormStateInterface $form_state, $f
...
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function commencement_core_node_view(array &$build, EntityInterface $node, EntityViewDisplayInterface $display, $view_mode) {
+  if ($node->bundle() === 'event' && $view_mode === 'full') {
+
+    // Ceremony information heading.
+    if ($display->getComponent('field_event_livestream_heading')) {
+      $build['field_event_livestream_heading'] = [
+        '#type' => 'markup',
+        '#markup' => '<h2 id="livestream" class="headline block__headline headline block__headline headline--left">' . t("Livestream") . '</h2>',
```
Comment (pyrello) [inline docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module]:
```suggestion
        '#markup' => '<h2 id="livestream" class="headline block__headline headline block__headline headline--left">' . t('Livestream') . '</h2>',
```
FWIW, I think the entire markup string can be passed to `t()`.

File: docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module [hunk 2]
```diff
@@ -86,6 +120,44 @@ function commencement_core_form_alter(&$form, FormStateInterface $form_state, $f
...
+      ];
+    }
+
+    // Ceremony information heading.
+    if ($display->getComponent('field_event_ceremony_info_heading')) {
+      $build['field_event_ceremony_info_heading'] = [
+        '#type' => 'markup',
+        '#markup' => '<h2 id="ceremony_info" class="headline block__headline headline block__headline headline--left">' . t("Ceremony Information") . '</h2>',
```
Comment (pyrello) [inline docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module]:
```suggestion
        '#markup' => '<h2 id="ceremony_info" class="headline block__headline headline block__headline headline--left">' . t('Ceremony Information') . '</h2>',
```

File: docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module [hunk 3]
```diff
@@ -86,6 +120,44 @@ function commencement_core_form_alter(&$form, FormStateInterface $form_state, $f
...
+
+    // Presidential seal.
+    if ($display->getComponent('field_event_presidents_seal')) {
+      $build['field_event_presidents_seal'] = [
+        '#type' => 'markup',
+        '#markup' => '<img
+          loading="lazy"
+          src="/themes/custom/uids_base/assets/images/Seal-Black-edge.svg"
+          alt="The Presidential seal of the University of Iowa."
+          class=""
```
Comment (pyrello) [inline docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module]:
```suggestion
          alt="The Presidential seal of the University of Iowa."
```
Not sure why this is included...

File: docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module [hunk 4]
```diff
@@ -186,6 +264,8 @@ function commencement_core_preprocess_field__field_media_file(&$variables) {
  * Implements hook_preprocess_HOOK().
  */
 function commencement_core_preprocess_field(&$variables) {
+  $node = $variables["element"]["#object"];
```
Comment (pyrello) [inline docroot/sites/commencement.uiowa.edu/modules/commencement_core/commencement_core.module]:
```suggestion
  $node = $variables['element']['#object'];
```

**Comment by joewhitsitt** [comment]
pulling to test

**Comment by joewhitsitt** [comment]
@GaryRidgway 

I am noticing some issues with the sidebar:

- Session program label seems to be inconsistent. First it showed up as Order of Events, now I am only getting the bare filename.
- Website is outputting the label. Example: https://commencement.uiowa.ddev.site/ceremony/spring-2024/graduate-masters-commencement-ceremony
- Wasn't there EoD talk about making the links in the sidebar look like links? There is no underline, color change, or hover effect.

---
Screenshot 2024-05-28 at 10 52 03 AM

---
Screenshot 2024-05-28 at 11 00 00 AM
---

I can't remember if the headings should show or not if there isn't content for them? Disregard if this is as expected.

Screenshot 2024-05-28 at 10 50 22 AM

**Comment by GaryRidgway** [comment]
> @GaryRidgway
> 
> I am noticing some issues with the sidebar:
> Session program label seems to be inconsistent. First it showed up as Order of Events, now I am only getting the bare filename.

Thanks for looking at this Joe. And you had uploaded the Program PDF on the session taxonomy, right?


> Website is outputting the label. Example: https://commencement.uiowa.ddev.site/ceremony/spring-2024/graduate-masters-commencement-ceremony

Good catch!

> Wasn't there EoD talk about making the links in the sidebar look like links? There is no underline, color change, or hover effect.

We had, however our default implementation is all black on the sidebar

> I can't remember if the headings should show or not if there isn't content for them? Disregard if this is as expected.

we had decided not to worry about the headings displaying or not displaying for now.

**Comment by bspeare** [comment]
>     * Wasn't there EoD talk about making the links in the sidebar look like links? There is no underline, color change, or hover effect.

The default behavior of the card links are black due to the how the card handles background colors: https://sandbox.prod.drupal.uiowa.edu/people/sean-adams-hiett.    We can either process in the `bg--none` class like on https://github.com/uiowa/uiowa/blob/main/docroot/sites/its.uiowa.edu/modules/its_core/its_core.module#L522 or shift our default for this layout to `bg--none` but that would require some greater refactoring and testing.

**Comment by joewhitsitt** [comment]
>And you had uploaded the Program PDF on the session taxonomy, right?

Yes. Went to the session term page and added an existing pdf file to fill the field and went back to the event node. one might have been weird cache blip, but the plain filename output doesn't seem right.

**Comment by joewhitsitt** [comment]
>The default behavior of the card links are black due to the how the card handles background colors

Dunno why I just started to notice this. I am in favor of consistency and making it easier to differentiate a link from text. Do we have an issue for this already @bspeare or should I make a new one?

**Comment by bspeare** [comment]
@joewhitsitt a new issue makes sense.

**Comment by bspeare** [comment]
Since we can control this, should this just be a link instead? `<a href="https://grad.uiowa.edu/commencement">Website</a>`?

Screenshot 2024-05-28 at 1 18 43 PM

Also, we've been using `fa-desktop` as our website link icon on the platform like at https://sandbox.prod.drupal.uiowa.edu/people/sean-adams-hiett.

**Comment by GaryRidgway** [comment]
> Since we can control this, should this just be a link instead? `<a href="https://grad.uiowa.edu/commencement">Website</a>`?
> 
> Screenshot 2024-05-28 at 1 18 43 PM
> Also, we've been using `fa-desktop` as our website link icon on the platform like at https://sandbox.prod.drupal.uiowa.edu/people/sean-adams-hiett.

good idea, I can change these

**Review by joewhitsitt — APPROVED** [review]
Thanks for making the additional changes

Screenshot 2024-05-28 at 3 27 38 PM

**Comment by pyrello** [comment]
> I can't remember if the headings should show or not if there isn't content for them? Disregard if this is as expected.

We decided to not worry about this functionality for now. Its rather complex to get one block to display based on the presence of multiple other blocks.

**Pull request merged**